### PR TITLE
feat: Add updateResource endpoint and tests [DEV-4606]

### PR DESCRIPTION
**Pull request by sownak**
- Add updateResource endpoint
- Add playwright tests

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-4606 Update Cheqd registrar /updateResouce operation](https://app.clickup.com/t/6600954/DEV-4606)

File: src/controllers/resource.ts [hunk 1]
```diff
@@ -37,14 +43,41 @@ export class ResourceController {
...
+		check('content')
+			.exists()
+			.isArray()
+			.custom((value) => {
+				if (value.length !== 1) return false;
+				if (typeof value[0] !== 'string') return false;
+				return true;
+			})
+			.withMessage('The content array must be provided and must have exactly one string'),
+		check('relativeDidUrl').exists().isString().contains('/resources/').withMessage(Messages.InvalidDidUrl),
```
Comment (DaevMithran) [inline src/controllers/resource.ts:74]: Can we make this optional relativeDidUrl 

if not provided we update the latest resource, by fetching the resource id using the name and type

File: src/controllers/resource.ts [hunk 2]
```diff
@@ -242,6 +275,146 @@ export class ResourceController {
...
+				return response
+					.status(400)
+					.send(Responses.GetInvalidResourceResponse(did, {}, secret, Messages.DidNotFound));
+			} else {
+				const didUrlIndex = resolvedDocument.didDocumentMetadata.linkedResourceMetadata.findIndex(
+					(resource: { resourceURI: string }) => resource.resourceURI === did + relativeDidUrl
+				);
+				if (didUrlIndex !== -1) {
+					existingResource = resolvedDocument.didDocumentMetadata.linkedResourceMetadata[didUrlIndex];
+					if (existingResource.resourceName !== name || existingResource.resourceType !== type)
```
Comment (DaevMithran) [inline src/controllers/resource.ts:338]: also validate if nextVersionId is null. Because only the latest resource can be updated

**Review by DaevMithran — CHANGES_REQUESTED** [review]

File: src/controllers/resource.ts [hunk 3]
```diff
@@ -37,14 +43,41 @@ export class ResourceController {
 				return true;
 			})
 			.withMessage('name, type and content are required'),
+		check('content').exists().isString().withMessage(Messages.Invalid),
```
Comment (DaevMithran) [inline src/controllers/resource.ts:46]: `content` should be optional

**Review by DaevMithran — CHANGES_REQUESTED** [review]

File: src/controllers/resource.ts (hunk 1 above)
Comment (sownak) [reply src/controllers/resource.ts:74]: Get using ?resourceName=Score%20schema-default&resourceType=anonCredsCredDef&resourceMetadata=true
Validate using the nextVersionId=null

**Comment by sownak** [comment]
@DaevMithran All comments are address in this PR

**Review by DaevMithran — APPROVED** [review]

**Pull request merged**